app/tests/spec/lib/crypto/scoped-keys.js

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  const { assert } = require('chai');
+  const sinon = require('sinon');
+  const ScopedKeys = require('lib/crypto/scoped-keys');
+
+  describe('lib/crypto/scoped-keys', () => {
+    const keys = {
+      kA: 'bba2ea983743324201a921e816f2e00e25da54473c9aa3ef050209c0f3bb8d86',
+      kB: 'f5c47b97aecaf7dca9e020e4ea427f8431334a505cda40f09f3d9577e0006185'
+    };
+    const scope = 'https://identity.mozilla.org/apps/notes';
+    const clientKeyData = {
+      [scope]: {
+        identifier: scope,
+        keyRotationSecret: '0000000000000000000000000000000000000000000000000000000000000000',
+        keyRotationTimestamp: 1510011454564
+      }
+    };
+    const derivedKey = {
+      k: '6pXqtKRy9VJ2aXc9cQsNjdB1058MMEOtfSmYL8TFTgo',
+      kid: '1510011455-A19NCXeloCwS_-QqRdgqMQ',
+      kty: 'oct',
+      scope: scope
+    };
+    const clientScopedKey = {
+      [scope]: derivedKey
+    };
+    const keysJwk = 'eyJrdHkiOiJFQyIsImtpZCI6IjVEakVLQ1ZSRGtCUFBLVTc4ZjNQOW92eU5EeDhnb1NWbGh0QzhFMlJfZXciLCJjcnYiOiJQLTI1NiIsIngiOiIzTXkwZzBNN3JwX2MyemMxNVlZM2xKcjlKcURrSmFXQjhLcTJ6aFhRTldNIiwieSI6IlVGZ05UVGVRbWlZTEE5VzJVTmIyemFaVHhzWHVtYnVpbDFhT0xlY1gxRk0ifQ'; //eslint-disable-line max-len
+
+    describe('_deriveScopedKeys', () => {
+      it('derives a key', () => {
+        return ScopedKeys._deriveScopedKeys(keys.kB, clientKeyData[scope]).then((derivedObject) => {
+          assert.deepEqual(derivedObject, derivedKey);
+        });
+      });
+
+      it('throws if no inputKey', () => {
+        return ScopedKeys._deriveScopedKeys().then(null, (err) => {
```

Comment:
With these `then(null`, if the error path is not taken, we'll never know. Can you change `null` to `assert.fail` both here and on line 50?